feedback/invalid_token.go

```diff
@@ -145,23 +144,29 @@ func (i *InvalidTokenHandler) processMessages() {
 	flushTicker := time.NewTicker(i.flushTime)
 	defer flushTicker.Stop()
 
-	for i.run {
+	for {
 		select {
 		case tk, ok := <-i.InChan:
 			if ok {
+				i.BufferLock.Lock()
 				i.Buffer = append(i.Buffer, tk)
+				i.BufferLock.Unlock()
 
+				i.BufferLock.Lock()
 				if len(i.Buffer) >= i.bufferSize {
 					l.Debug("buffer is full")
 					i.deleteTokens(i.Buffer)
 					i.Buffer = make([]*InvalidToken, 0, i.bufferSize)
 				}
+				i.BufferLock.Unlock()
 			}
 
 		case <-flushTicker.C:
 			l.Debug("flush ticker")
 			i.deleteTokens(i.Buffer)
```

Comment:
shouldn't we guard `deleteTokens` under the `.Lock()` since we're changing `i.Buffer`?

Reply:
True. Good catch!